gomanifest/internal/manifest.go

```diff
@@ -0,0 +1,135 @@
+package internal
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+)
+
+/* MANIFEST FILE VERSION */
+const manifestVersion = "v1"
+
+// Dependency ... Direct and transitive dependency details.
+type Dependency struct {
+	Name         string       `json:"package"`
+	Version      string       `json:"version"`
+	Dependencies []Dependency `json:"deps,omitempty"`
+}
+
+// Manifest ... Final manifest file structure
+type Manifest struct {
+	Version  string       `json:"version"`
+	Main     string       `json:"main"`
+	Packages []Dependency `json:"packages"`
+}
+
+// transformVersion ... Converts the golang version string into semver without 'v' and appended text after '+'
+func transformVersion(inVersion string) string {
+	var outVersion string = strings.Replace(inVersion, "v", "", 1)
+
+	return strings.Split(outVersion, "+")[0]
+}
+
+// getPackageName ... Utility function to convert package + module data into package name used by manifest.
+func getPackageName(depPackage DepPackage) string {
+	// Get module / package@module
+	if depPackage.ImportPath != depPackage.Module.Path {
+		return depPackage.ImportPath + "@" + depPackage.Module.Path
+	}
+
+	return depPackage.ImportPath
+}
+
+// newDirectDependency ... Return a new direct depenency for a given go package.
+func newDirectDependency(depPackage DepPackage, depPackages map[string]DepPackage) Dependency {
```

Comment:
Use pointer for depPackages.,
Ditto else where.
```suggestion
func newDirectDependency(depPackage DepPackage, depPackages *map[string]DepPackage) Dependency {
```

Reply:
This is done.